### PR TITLE
Enable strict globally, fix comparison of IntPtr, Guid with null

### PR DESCRIPTION
**Pull request by danmosemsft**
Relates to https://github.com/dotnet/corefx/issues/31447 and https://github.com/dotnet/corefx/issues/31456

```c#
            IntPtr ptr = IntPtr.Zero;
            Console.WriteLine(ptr == null);   // false
            Console.WriteLine(ptr != null);    // true
```
IntPtr should never be compared with null.

I had a cursory look at other uses of GetProcAddress and they all seem to be comparing to IntPtr.Zero as they should, but a Roslyn analyzer would probably be necessary to be sure.

### CODE REVIEW

**Comment by jjxtra** [comment]
This fixes this one piece of code, are there any other framework pieces that do this, and of course what about the billions of lines of C# code out in the wild? See issue #31456

**Comment by sharwell** [comment]
> but a Roslyn analyzer would probably be necessary to be sure.

As mentioned later in https://github.com/dotnet/corefx/issues/31456#issuecomment-408646070, the C# compiler actually supports this analysis but it's not enabled in corefx yet.

**Comment by EgorBo** [comment]
@danmosemsft a few more minor cases:

- https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/HandleTests.cs#L19 (Value is `IntPtr`)
- https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/StandardOleMarshalObject.cs#L32 

(found by a small Roslyn script)

**Comment by danmosemsft** [comment]
@EgorBo thanks, fixed.
@4creators not sure whether you were going to do corefx, so I just rolled it into this PR.

File: src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj [hunk 1]
```diff
@@ -3,6 +3,11 @@
     <ProjectGuid>{82B54697-0251-47A1-8546-FC507D0F3B08}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Configurations>netcoreapp-Debug;netcoreapp-Release;netstandard-Debug;netstandard-Release</Configurations>
+    <!-- 
+    We wish to test operations that would result in
+    "Operator '-' cannot be applied to operands of type 'ushort' and 'EnumArithmeticTests.UInt16Enum'"
+    -->
+    <Features>$(Features.Replace('strict', '')</Features>
```
Comment (danmosemsft) [inline src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]: https://github.com/dotnet/roslyn/issues/28935

**Comment by danmosemsft** [comment]
@pakrym could you OK the pipelines change? Seems innocuous

**Comment by pakrym** [comment]
Yes, seems alright.

**Comment by danmosemsft** [comment]
@dotnet/dnceng why am I not seeing all the CI legs broken out above even though it's not cmopleted?
image

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs [hunk 2]
```diff
@@ -1229,7 +1229,6 @@ private static ulong TransferDataFromRequestStream(IntPtr buffer, int length, Ea
                     asyncRead = easy._requestContentStream.ReadAsync(
                        new Memory<byte>(sts.Buffer, 0, Math.Min(sts.Buffer.Length, length)), easy._cancellationToken);
                 }
-                Debug.Assert(asyncRead != null, "Badly implemented stream returned a null task from ReadAsync");
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs]: Why was this deleted?

File: src/System.Transactions.Local/src/System.Transactions.Local.csproj [hunk 3]
```diff
@@ -4,6 +4,11 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Configurations>netcoreapp-Debug;netcoreapp-Release;uap-Windows_NT-Debug;uap-Windows_NT-Release</Configurations>
+    <!-- 
+    There are many instances of comparing an instance of a struct with null. Until these are all fixed,
+    disable the diagnostic.
```
Comment (stephentoub) [inline src/System.Transactions.Local/src/System.Transactions.Local.csproj]: Issue tracking it?

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs (hunk 2 above)
Comment (sharwell) [reply src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs]: `ValueTask<T>` is a struct, so it's impossible for a method to return a null one. However, if `default(ValueTask<int>)` cannot be awaited (or is otherwise not valid), it seems like an assertion could say but have a different form.

File: src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj [hunk 4]
```diff
@@ -3,6 +3,11 @@
     <ProjectGuid>{82B54697-0251-47A1-8546-FC507D0F3B08}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Configurations>netcoreapp-Debug;netcoreapp-Release;netstandard-Debug;netstandard-Release</Configurations>
+    <!-- 
+    We wish to test operations that would result in
+    "Operator '-' cannot be applied to operands of type 'ushort' and 'EnumArithmeticTests.UInt16Enum'"
```
Comment (sharwell) [inline src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]: ❔ Can you not use targeted warning suppressions to allow the specific cases?

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs]: So, I didn't see the type here. It's fine to remove the assert. This is likely left over from when it was a Task.

File: src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj (hunk 4 above)
Comment (danmosemsft) [reply src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]: I could not get #pragma disable 0019 to work. Does it work for you?

**Comment by mmitche** [comment]
@danmosemsft I'm not sure what you mean.  I see the legs broken out for that commit (all 14) as expected.

**Comment by danmosemsft** [comment]
@mmitche they are now but as you see in the screenshot they were not earlier..this was part way through the run.

**Comment by mmitche** [comment]
@danmosemsft Did you have a refreshed page?  All the legs started at the exact same time.  Occasionally the UI doesn't update without a page refresh though.

**Comment by dagood** [comment]
When you click merge (and before you confirm the merge), you don't see the CI status anymore. Hitting cancel brings you back to the normal view. Could this be what you were seeing?

File: src/System.Transactions.Local/src/System.Transactions.Local.csproj (hunk 3 above)
Comment (danmosemsft) [reply src/System.Transactions.Local/src/System.Transactions.Local.csproj]: #31476
